File: csharp/DVSClient/DVSClient/address/format/AddressFormattedLine.cs
using DVSClient.Common;
using DVSClient.Server.Address.Format;

namespace DVSClient.Address.Format
{
    public class AddressFormattedLine
    {
        internal string label { get; }
        internal string line { get; }
        internal bool hasOverflownToOtherLine { get; }
        internal bool isTruncated { get; }
        internal LineContent? lineContent { get; }

        public AddressFormattedLine(RestApiAddressFormattedLine result)
        {
            label = result.Label ?? string.Empty;
            line = result.Line ?? string.Empty;
            hasOverflownToOtherLine = result.HasOverflownToOtherLine ?? false;
            isTruncated = result.IsTruncated ?? false;
            lineContent = result.LineContent?.GetEnumValueFromJsonName<LineContent>();
        }
    }
}

File: typescript/src/address/format/addressFormattedLine.ts
import { RestApiAddressFormattedLine } from "../../server/address/format/restApiAddressFormattedLine";
import { LineContent, lookupLineContent } from "./lineContent";

export type AddressFormattedLine = {
    label: string;
    line: string;
    hasOverflownToOtherLine: boolean;
    isTruncated: boolean;
    lineContent?: LineContent;
};

export function restApiToAddressFormattedLine(response: RestApiAddressFormattedLine): AddressFormattedLine {
    return {
        label: response.label ?? "",
        line: response.line ?? "",
        hasOverflownToOtherLine: response.has_overflown_to_other_line ?? false,
        isTruncated: response.is_truncated ?? false,
        lineContent: lookupLineContent(response.line_content)
    }
}